pkg/dbal/DbalConsumerHelperTrait.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Enqueue\Dbal;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\ParameterType;
+use Doctrine\DBAL\Types\Type;
+
+trait DbalConsumerHelperTrait
+{
+    abstract public function getContext(): DbalContext;
+
+    abstract public function getConnection(): Connection;
+
+    protected function fetchMessage(array $queues, string $deliveryId, int $redeliveryDelay): ?array
```

Comment:
What if this method returns an instance of `DbalMessage`. You have access to context so you can use a `$context->convertMessage()` method.